extensions-contrib/materialized-view-selection/src/main/java/io/druid/query/materializedview/MaterializedViewUtils.java

```diff
@@ -0,0 +1,282 @@
+    } else if (query instanceof GroupByQuery) {
+      GroupByQuery q = (GroupByQuery) query;
+      dimensions.addAll(extractFieldsFromAggregations(q.getAggregatorSpecs()));
+      for (DimensionSpec spec : q.getDimensions()) {
+        String dim = spec.getDimension();
+        dimensions.add(dim);
+      }
+    }
+    return dimensions;
+  }
+
+  private static Set<String> extractFieldsFromAggregations(List<AggregatorFactory> aggs) 
+  {
+    Set<String> ret = new HashSet<>();
+    for (AggregatorFactory agg : aggs) {
+      if (agg instanceof FilteredAggregatorFactory) {
+        FilteredAggregatorFactory fagg = (FilteredAggregatorFactory) agg;
+        ret.addAll(getDimensionsInFilter(fagg.getFilter()));
+      }
+      ret.addAll(agg.requiredFields());
+    }
+    return ret;
+  }
+
+  private static Set<String> getDimensionsInFilter(DimFilter dimFilter)
+  {
+    if (dimFilter instanceof AndDimFilter) {
+      AndDimFilter d = (AndDimFilter) dimFilter;
+      Set<String> ret = new HashSet<>();
+      for (DimFilter filter : d.getFields()) {
+        ret.addAll(getDimensionsInFilter(filter));
+      }
+      return ret;
+    } else if (dimFilter instanceof OrDimFilter) {
+      OrDimFilter d = (OrDimFilter) dimFilter;
+      Set<String> ret = new HashSet<>();
+      for (DimFilter filter : d.getFields()) {
+        ret.addAll(getDimensionsInFilter(filter));
+      }
+      return ret;
+    } else if (dimFilter instanceof NotDimFilter) {
+      NotDimFilter d = (NotDimFilter) dimFilter;
+      return getDimensionsInFilter(d.getField());
+    } else if (dimFilter instanceof BoundDimFilter) {
+      BoundDimFilter d = (BoundDimFilter) dimFilter;
+      return Sets.newHashSet(d.getDimension());
+    } else if (dimFilter instanceof InDimFilter) {
+      InDimFilter d = (InDimFilter) dimFilter;
+      return Sets.newHashSet(d.getDimension());
+    } else if (dimFilter instanceof IntervalDimFilter) {
+      IntervalDimFilter d = (IntervalDimFilter) dimFilter;
+      return Sets.newHashSet(d.getDimension());
+    } else if (dimFilter instanceof LikeDimFilter) {
+      LikeDimFilter d = (LikeDimFilter) dimFilter;
+      return Sets.newHashSet(d.getDimension());
+    } else if (dimFilter instanceof RegexDimFilter) {
+      RegexDimFilter d = (RegexDimFilter) dimFilter;
+      return Sets.newHashSet(d.getDimension());
+    } else if (dimFilter instanceof SearchQueryDimFilter) {
+      SearchQueryDimFilter d = (SearchQueryDimFilter) dimFilter;
+      return Sets.newHashSet(d.getDimension());
+    } else if (dimFilter instanceof SelectorDimFilter) {
+      SelectorDimFilter d = (SelectorDimFilter) dimFilter;
+      return Sets.newHashSet(d.getDimension());
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * interval2 - interval1
+   * 
+   * @param interval2
+   * @param interval1
+   * @return results of interval2-interval1
+   */
+  public static List<Interval> minus(List<Interval> interval2, List<Interval> interval1)
```

Comment:
Would you please add more detailed javadoc for this method? It should have at least, 1) what parameters are, 2) what this method does, 3) what is the meaning of `minus`, and 4) what the assumptions about the input parameters are.